scripts/generator/graphql_generator/csharp/SDK/Editor/UnityEditorLoader.cs.erb

```diff
@@ -0,0 +1,80 @@
+#if !SHOPIFY_MONO_UNIT_TEST
+namespace <%= namespace %>.SDK.Editor {
+    using Shopify.Unity.SDK;
+    using System.Collections;
+    using System.Collections.Generic;
+    using UnityEditor;
+    using UnityEngine;
+    using System.Text;
+
+    /// <summary>
+    /// A Query Loader built to load queries in editor scripts, when the normal game loop is not available.
+    /// </summary>
+    public class UnityEditorLoader : BaseLoader {
+        public UnityEditorLoader(string domain, string accessToken) : base(domain, accessToken) {
+            EditorApplication.update += Update;
+        }
+
+        ~UnityEditorLoader() {
+            EditorApplication.update -= Update;
+        }
+
+        #region Interface
+
+        public override void Load(string query, LoaderResponseHandler callback) {
+            var body = Encoding.ASCII.GetBytes(query);
+            var w = new WWW(Url, body, _headers);
+            _activeRequests.Add(ProcessRequest(w, callback));
+        }
+
+        public override string SDKVariantName() {
+            return "unity";
+        }
+
+        public override void SetHeader(string key, string value) {
+            _headers.Add(key, value);
+        }
+
+        # endregion
+
+        #region Implementation
+
+        private List<IEnumerator> _activeRequests = new List<IEnumerator>();
+
+        private Dictionary<string,string> _headers = new Dictionary<string,string>();
+
+        private string Url {
+            get {
+                return "https://" + Domain + "/api/graphql.json";
+            }
+        }
+
+        private IEnumerator ProcessRequest(WWW w, LoaderResponseHandler callback) {
+            // We have to use this enumerator approach because the www class inherits from CustomYieldInstruction,
+            // which isn't well supported in the editor.
+
+            while(!w.isDone) {
+                yield return null;
+            }
+
+            if (!string.IsNullOrEmpty(w.error)) {
+                callback(null, w.error);
+            } else {
+                callback(w.text, null);
+            }
+
+            w.Dispose();
+        }
+
+        private void Update() {
+            foreach(var request in new List<IEnumerator>(_activeRequests)) {
+                if (!request.MoveNext()) {
```

Comment:
This is basically just removing the last element?

Reply:
no, the elements in this list are essentially coroutines. In unity, coroutines are just an IEnumerator where MoveNext returns true if it should keep running or false if it is finished. the WWW will start a background download, so we just need to wait until www.isDone is true, the coroutine that wraps the download and is stored in activeRequests will yield true for MoveNext until the function exits, in which case we dont need to tick the reference anymore so we can just remove it from the list of references we need to tick.